src/fortran/fdb.f90

```diff
@@ -1,39 +1,329 @@
+      CHARACTER(len=*),INTENT(INOUT)  :: s
+      INTEGER :: i
+      DO i = 1,SIZE(a)
+         s(i:i) = 'z' ! a(i) ! FAILS HERE
+      END DO
+   END SUBROUTINE copy_a2s
+
+   FUNCTION index_chararray(a,s) result(idx)  
+      CHARACTER,INTENT(IN) :: a(:)
+      CHARACTER(len=*), INTENT(IN)  :: s
+      CHARACTER :: s_a(LEN(s))
+      INTEGER  :: i, j, idx
+      logical :: equal
+      DO i = 1,LEN(s)
+         s_a(i) = s(i:i)
+      END DO
+      DO i = 1,SIZE(a)-SIZE(s_a)
+         DO j = 1, SIZE(s_a)
+            equal = a(i+j-1) == s_a(j)
+            IF (.not. equal) THEN
+               exit
+            END IF
+         END DO
+         IF (equal) THEN
+            idx=i
+            exit
+         END if
+      END DO
+      IF (.not. equal) THEN
+         idx=0
+      END IF
+   END FUNCTION index_chararray
+   
+   SUBROUTINE add_key(key, igrib, keyname_str, type)  
+      use, intrinsic :: iso_c_binding, only : c_int, c_ptr, c_char
+      type(c_ptr), INTENT(INOUT)                  :: key
+      character(len=*), INTENT(IN)                :: keyname_str
+      character(len=*), INTENT(IN), OPTIONAL      :: type
+      ! Local vars
+      character(kind=c_char), dimension(128)      :: keyname
+      character(kind=c_char), dimension(128)      :: keyvalue
+      character(len=128)                          :: keyvalue_str
+      integer                                     :: keyvalue_int
+      integer, INTENT(IN)                         :: igrib
+      integer                                     :: res
+      if (type == 'integer') then
+         call get_value_of_key(igrib, trim(keyname_str), keyvalue_str, keyvalue_int) 
+      else
+         call get_value_of_key(igrib, trim(keyname_str), keyvalue_str) 
+      end if
+      call copy_s2a(keyvalue, trim(keyvalue_str))
+      call copy_s2a(keyname, trim(keyname_str))
+      res = fdb_key_add(key, keyname, keyvalue)
+   END SUBROUTINE add_key
+
+
+   SUBROUTINE convertValues(numStrings, values_str_array, values_array, values_ptr)
+      use, intrinsic :: iso_c_binding 
+      integer(kind=c_int) , INTENT(IN)              :: numStrings
+      CHARACTER(len=*), INTENT(IN) :: values_str_array(numStrings)
+      CHARACTER(kind=c_char), TARGET , INTENT(INOUT)  :: values_array(32,numStrings)
+      TYPE(C_PTR), INTENT(OUT)                 :: values_ptr(numStrings)
+      integer                                  :: ns, i
+      character(len=32)                        :: value_str
+      character(kind=c_char), dimension(32)    :: value
+      character(len=5)                         :: ns_str
+      DO ns = 1, numStrings
+         ! cehck that values_str_array(ns) is not greater in size than 32, or hard coded value.
+         call copy_s2a(values_array(:,ns), trim(values_str_array(ns)))
+         values_ptr(ns) = C_LOC(values_array(:,ns))
+      END DO
+            
+   END SUBROUTINE
+
+
+   SUBROUTINE fdb_request_add_fortran(req, keyname_str, values_str_array)
```

Comment:
I would rename to fdb_request_add_values (values, since it can add a sequence of values, and also fortran is needed, since the subroutine is already in fortran

Reply:
Renamed fdb_request_add_fortran to fdb_request_add_values